### DISCUSSION TITLE
SQL Exporter Failing to Connect to SQL Server Named Instance

### DISCUSSION THREAD

**Discussion by aszkuat** [burningalchemist/sql_exporter#515] [Q&A] [answered]
We are experiencing persistent issues with the SQL Exporter failing to gather metrics from our SQL Server named instance. Despite multiple attempts to configure the data source name correctly, the exporter either times out or returns errors indicating invalid URL escape sequences.
Environment:

SQL Exporter version: 0.14.3
SQL Server instance: 
server01.domain.local\instance1
server01.domain.local\instance2
Authentication: SQL authentication using username and password

We use dynamic ports for MS SQL.

Tried different formats for the data source name:
data_source_name: sqlserver://username:password@server01.domain.local?database=instance1&encrypt=false&trustServerCertificate=true
data_source_name: 'sqlserver://username:password@server01.domain.local\instance1'
data_source_name: 'sqlserver://username:password@server01.domain.local\\instance1' 

What should the data source field look like when not using static ports?

**Comment by burningalchemist** [comment, marked as answer]
Hi @aszkuat, sql_exporter is using `dburl` library for unified DSN connection string format across many databases, so I'd suggest to try this syntax out as per their documentation:

```
sqlserver://user:pass@remote-host.com/instance/dbname
```

where instance is `instance1` and `dbname` is your desired database name to connect to. `dbname` might also be a mandatory value (whereas `instance` can be optional), but it's controlled by the mssql driver, so omitting might potentially work. Give it a try.

Here is the reference [link](https://github.com/xo/dburl?tab=readme-ov-file#database-connection-url-overview).

Also, the reference from the driver [here](https://github.com/microsoft/go-mssqldb?tab=readme-ov-file#the-connection-string-can-be-specified-in-one-of-three-formats).

**Comment by aszkuat** [comment]
Thank you, the option works.
